# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: caps/nsIPrincipal.idl
# Commit: 3ce54e39ebcd
# Full Hash: 3ce54e39ebcdf069615571d1805af75ad969f006
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-06-03 04:20:58
# Regressor Bug: 1638358
# File Overlap Count: 1
# Description:
#   Bug 1638358 - Cookie Schemeful Same-Site - part 5 - schemeful comparison, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D75629
# ==============================================================================

diff -r dff1c1aca50c -r 3ce54e39ebcd caps/nsIPrincipal.idl
--- a/caps/nsIPrincipal.idl	Tue Jun 02 13:49:45 2020 +0000
+++ b/caps/nsIPrincipal.idl	Tue Jun 02 13:50:12 2020 +0000
@@ -287,7 +287,12 @@
     /* Returns the Spec of the Principals URI with
     * user/pass/ref/query stripped for privacy and spoof prevention
     */
-     readonly attribute ACString exposableSpec;
+    readonly attribute ACString exposableSpec;
+
+    /**
+     * Return the scheme of the principals URI
+     */
+    readonly attribute ACString scheme;
 
     /**
     * Checks if the Principal's URI Scheme matches with the parameter